programs/axelar-solana-gateway/src/processor/call_contract.rs

```diff
@@ -58,32 +58,37 @@ impl Processor {
             GatewayConfig::read(&data).ok_or(GatewayError::BytemuckDataLenInvalid)?;
         assert_valid_gateway_root_pda(gateway_config.bump, gateway_root_pda.key)?;
 
-        // compute the payload hash
-        let payload_hash = solana_program::keccak::hash(payload).to_bytes();
-
-        // Check: sender is a program
-        if sender.executable {
-            // If so, check that signing PDA is valid
-            let expected_signing_pda =
-                create_call_contract_signing_pda(*sender.key, signing_pda_bump)?;
+        if sender.is_signer {
+            // Direct signer, so not a program, continue
+        } else {
```

Comment:
```suggestion
        if !sender.is_signer {
```

Any particular reason not to do this?

Reply:
Only subjective reasons, I found it easier to grasp presented this way.